### PR TITLE
remove factories/rename authors

### PR DESCRIPTION
**Pull request by willydouhard**
This is a breaking change. I propose to remove the factory / run / postprocess paradigm as it is not as meaningful as when we introduced them (we automate less things nowadays).

It was confusing for the users. We do not lose any feature though, everything is doable with `on_chat_start`, `on_message` and `user_session`.  The real integrations are at the callback level anyways (which are now better tested).

`langchain_rename` is now `author_rename` and works for every use case.

It also simplifies the code base.

### CODE REVIEW

File: cypress/support/utils.ts [hunk 1]
```diff
@@ -50,22 +50,26 @@ export async function runTest(test: string) {
         console.log("Running with local db");
       }
 
-      return new Promise(async (resolve, reject) => {
+      const promise = new Promise(async (resolve, reject) => {
```
Comment (tpatel) [inline cypress/support/utils.ts:53]: We could go a little bit simpler

```suggestion
      await new Promise(async (resolve, reject) => {
```

File: cypress/support/utils.ts [hunk 2]
```diff
@@ -50,22 +50,26 @@ export async function runTest(test: string) {
...
+              resolve(true);
             } else {
               console.log("Process killed successfully");
               resolve(true);
             }
           });
         }
       });
+
+      await promise;
```
Comment (tpatel) [inline cypress/support/utils.ts:72]: And remove this

```suggestion
```

**Review by tpatel — APPROVED** [review]
✨✨✨ AWESOME ✨✨✨

I love the cleaner interface (and less code)

File: src/chainlit/lc/callbacks.py [hunk 3]
```diff
@@ -115,6 +115,23 @@ def check_if_answer_reached(self) -> bool:
...
+
+        streamed_message = Message(
+            author=author,
+            indent=indent,
+            llm_settings=llm_settings,
+            prompt=self.consume_last_prompt(),
+            content="",
+        )
+        self.stream = streamed_message
+
```
Comment (ramnes) [inline src/chainlit/lc/callbacks.py:134]: 😍

File: src/chainlit/socket.py [hunk 4]
```diff
@@ -98,31 +98,30 @@ def ask_user_fn(data, timeout):
...
-    if isinstance(session.auth_client, CloudAuthClient) and config.project.database in [
-        "local",
-        "custom",
-    ]:
-        await session.db_client.create_user(session.auth_client.user_infos)
-
-    if config.code.on_chat_start:
-        """Call the on_chat_start function provided by the developer."""
-        await config.code.on_chat_start()
+    try:
```
Comment (ramnes) [inline src/chainlit/socket.py:101]: let's do that in another PR and wrap all websocket handlers at once

**Review by ramnes — APPROVED** [review]
negative diffs are the best diffs

File: src/chainlit/socket.py (hunk 4 above)
Comment (willydouhard) [reply src/chainlit/socket.py:101]: actually I had to do that to debug 🤣
